tests/perf/perf_tests.cc

```diff
@@ -201,14 +210,25 @@ struct stdout_printer final : result_printer {
                "number of runs:", c.number_of_runs,
                "number of cores:", smp::count,
                "random seed:", c.random_seed);
-    fmt::print(header_format_string, "test", "iterations", "median", "mad", "min", "max", "allocs", "tasks", "inst");
+    fmt::print(fmt::runtime(header_format_string()), "test", "iterations", "median", "mad", "min", "max", "allocs", "tasks", "inst");
   }
 
   virtual void print_result(const result& r) override {
-    fmt::print(format_string, r.test_name, r.total_iterations / r.runs, duration { r.median },
+    fmt::print(fmt::runtime(format_string()), r.test_name, r.total_iterations / r.runs, duration { r.median },
                duration { r.mad }, duration { r.min }, duration { r.max },
                r.allocs, r.tasks, r.inst);
   }
+
+private:
+  std::string header_format_string() const {
+    return fmt::format("{{:<{}}} {{:>11}} {{:>11}} {{:>11}} {{:>11}} {{:>11}} {{:>11}} {{:>11}} {{:>11}}\n",
+        name_column_length());
+  }
+
+  std::string format_string() const {
+    return fmt::format("{{:<{}}} {{:>11}} {{:>11}} {{:>11}} {{:>11}} {{:>11}} {{:>11.3f}} {{:>11.3f}} {{:>11.1f}}\n",
+        name_column_length());
+  }
```

Comment:
You can write `format("{:<{}}", name, width)` to avoid the awkward runtime format strings.

Reply:
Oh nice, yes that's a lot cleaner. Have changed.